google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -594,9 +625,24 @@ public Builder setAge(Integer age) {
          * Sets the date a Blob should be created before for an Action to be executed. Note that
          * only the date will be considered, if the time is specified it will be truncated. This
          * condition is satisfied when an object is created before midnight of the specified date in
-         * UTC. *
+         * UTC.
+         *
+         * @deprecated Use {@link #setCreateBeforeOffsetDateTime(OffsetDateTime)}
          */
+        @Deprecated
         public Builder setCreatedBefore(DateTime createdBefore) {
+          if (createdBefore == null) return this;
+          return setCreateBeforeOffsetDateTime(
+              millisOffsetDateTimeCodec.encode(createdBefore.getValue()));
```

Comment:
Similar to the getter, I think this can be simplified to
```suggestion
          return setCustomTimeBeforeOffsetDateTime(
              Utils.dateTimeCodec.nullable().decode(customTimeBefore));
```